pkg/pillar/cmd/zfsmanager/zfsstoragemetrics.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2022 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package zfsmanager
+
+import (
+	"container/list"
+	"fmt"
+	"time"
+
+	libzfs "github.com/bicomsystems/go-libzfs"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/lf-edge/eve/pkg/pillar/vault"
+	"github.com/lf-edge/eve/pkg/pillar/zfs"
+)
+
+const (
+	storageMetricsPublishInterval = 1 * time.Second // interval between publishing metrics from zfs
+)
+
+func storageMetricsPublisher(ctxPtr *zfsContext) {
+	if vault.ReadPersistType() != types.PersistZFS {
+		return
+	}
+	collectAndPublishStorageMetrics(ctxPtr)
+
+	t := time.NewTicker(storageMetricsPublishInterval)
+	for {
+		select {
+		case <-t.C:
+			collectAndPublishStorageMetrics(ctxPtr)
+		}
+	}
+}
+
+func checkZFSPoolInMetrics(ctxPtr *zfsContext) error {
+	zpoolList, err := libzfs.PoolOpenAll()
+	if err != nil {
+		return fmt.Errorf("get zpool list failed %v", err)
+	} else {
+		for _, zpool := range zpoolList {
+			defer zpool.Close()
+			poolName, err := zpool.Name()
+			if err != nil {
+				return fmt.Errorf("get zpool name failed %v", err)
+			}
+
+			pub, _ := ctxPtr.storageMetricsPub.Get(poolName)
+			if pub == nil {
```

Comment:
If it was already published don't you need to re-publish it with the current values?